katello/hooks/pre/30-upgrade.rb

```diff
@@ -37,6 +37,48 @@ def migrate_foreman
   Kafo::Helpers.execute('foreman-rake db:migrate')
 end
 
+def postgresql_10_upgrade
+  return unless facts[:os][:release][:major] == '7'
+
+  #Ensure there is at least 1x /var/lib/pgsql free disk space available on the main filesystem
+  current_postgres_dir = '/var/lib/pgsql'
+  new_postgres_dir = '/var/opt/rh/rh-postgresql10/lib/pgsql'
+
+  begin
+    until File.directory?(new_postgres_dir)
+      directory = File.dirname(directory)
+    end
+
+    postgres_size = `du -k -s #{current_postgres_dir}`.split[0].to_i
+
+    if available_space(new_postgres_dir) < postgres_size
+      fail_and_exit 'The postgres upgrade requires at least #{postgres_size} KB of storage.'
+    end
+  rescue StandardError
+    fail_and_exit 'Failed to verify available disk space'
+  end
+
+  stop_services
+  Kafo::Helpers.execute('yum -y install rh-postgresql10-postgresql-server')
+  Kafo::Helpers.execute('scl enable rh-postgresql10 -- postgresql-setup --upgrade')
+  Kafo::Helpers.execute('yum -y remove postgresql postgresql-server')
+  Kafo::Helpers.execute('rm -f /etc/systemd/system/postgresql.service')
+  Kafo::Helpers.execute('yum -y install rh-postgresql10-syspaths')
+end
+
+def available_space(directory = nil)
+  directory = directory.nil? ? :/ : directory.to_sym
```

Comment:
I tackled the default and `to_sym` this way -- seemed to work for me testing locally.